activerecord/test/cases/migration/columns_test.rb

```diff
@@ -128,9 +128,9 @@ def test_remove_column_with_index
         add_column "test_models", :hat_name, :string
         add_index :test_models, :hat_name
 
-        assert_equal 1, connection.indexes("test_models").size
```

Comment:
There is something strange here, sqlite does remove the column but the index definition is not removed thus the test fails with the previous expectations

Reply:
This is confirmed to be a bug with sqlite https://sqlite.org/forum/forumpost/9daeccb4f5

Reply:
and fixed here: https://www.sqlite.org/src/info/1c24a659e6d7f3a1

Reply:
I'll update the PR using the fixed version of sqlite and restoring the check on the index for this test